wallet/wallet-controller/src/sync/mod.rs

```diff
@@ -82,50 +102,91 @@ pub async fn sync_once<T: NodeInterface>(
         let chain_info =
             rpc_client.chainstate_info().await.map_err(ControllerError::NodeCallError)?;
 
-        let (wallet_block_id, wallet_block_height) =
-            wallet.best_block().map_err(ControllerError::WalletError)?;
+        // first sync the unsynced account until it matches the other accounts
+        if let Some((wallet_block_id, wallet_block_height)) = wallet.best_block_unsynced_acc() {
```

Comment:
I think there is a bug. What would happen if there is a reorg while scanning the new account? This would always be `false`:
```
        if account.best_block().1
            == self
                .accounts
                .values()
                .next()
                .ok_or(WalletError::WalletNotInitialized)?
                .best_block()
                .1
        {
            self.unsynced_account
                .take()
                .map(|(index, acc)| self.accounts.insert(index, acc));
        }
```
Personally, I would prefer to keep things simple and sync each account independently.

Reply:
I think it should be fine, the unsynced account starts with genesis block, and the other accounts have already some blocks up to some block height X, this syncs the unsynced account up to that height X not comparing the block ID. If a reorg has happened the unsynced will now be on the new chain but on the same height, and after that, the rest of the accounts will also sync with the node.